subworkflows/ebi-metagenomics/fasta_domainannotation/meta.yml

```diff
@@ -0,0 +1,52 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/subworkflows/yaml-schema.json
+name: "fasta_domainannotation"
+description: Protein domain annotation
+keywords:
+  - protein
+  - domain
+  - annotation
+components:
+  - blast/makeblastdb
+  - blast/blastp
+  - diamond/makedb
+  - diamond/blastp
+  - interproscan
+  - eggnogmapper
+input:
+  - ch_fasta:
+      type: list
+      description: |
+        The input channel containing the meta id and the fasta with the proteins to annotate
+        Structure: [ val(meta), path(fasta) ]
+      pattern: "*.{fa/fasta/fa.gz/fasta.gz}"
+  - ch_blast_fasta:
+      type: file
+      description: |
+        The input channel containing the fasta with the protein database (e.g. UniprotSP) including their annotations, which blastp will be used against
```

Comment:
Even though this is agnostic of the reference DB, shouldn't UniprodSP be strongly recommended?